staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1115,12 +1115,7 @@ func (c *cacheWatcher) Stop() {
 	c.forget()
 }
 
-// TODO(#73958)
-// stop() is protected by Cacher.Lock(), rename it to
-// stopThreadUnsafe and remove the sync.Mutex.
-func (c *cacheWatcher) stop() {
-	c.Lock()
-	defer c.Unlock()
+func (c *cacheWatcher) stopThreadUnsafe() {
```

Comment:
1. Please add a comment to cacheWatcher structure that this is not-thread-safe.

2. Here please add a comment that we rely on the fact that stopThredunsafe is actually protected by Cacher.Lock()

3. Remove mutex from cacheWatcher structure.